packages/enzyme-test-suite/test/ReactWrapper-spec.jsx

```diff
@@ -6854,6 +6854,123 @@ describeWithDOM('mount', () => {
         wrapper.instance().setDeepDifferentState();
         expect(updateSpy).to.have.property('callCount', 1);
       });
+
+      describe('setProps calls `componentDidUpdate` when `getDerivedStateFromProps` is defined', () => {
+        class DummyComp extends PureComponent {
+          constructor(...args) {
+            super(...args);
+            this.state = { state: -1 };
+          }
+
+          static getDerivedStateFromProps({ changeState, counter }, prevState) {
+            return changeState ? { state: counter * 10 } : prevState;
```

Comment:
Shouldn't we return `null` when state does not need to be updated? https://reactjs.org/docs/react-component.html#static-getderivedstatefromprops

Reply:
> `getDerivedStateFromProps` is invoked right before calling the render method, both on the initial mount and on subsequent updates. It should return an object to update the state, or **null** to update nothing

Reply:
Yes, that's true. However, returning the previous state should result in state not being updated either. Updating to null still fails the same.